x/stakeibc/keeper/msg_server_restore_interchain_account.go

```diff
@@ -12,33 +12,29 @@ import (
 	"github.com/Stride-Labs/stride/v4/x/stakeibc/types"
 )
 
+// 6 function, 2 for verify record after exec, 4 for test (1 success, 3 for failure)
 func (k msgServer) RestoreInterchainAccount(goCtx context.Context, msg *types.MsgRestoreInterchainAccount) (*types.MsgRestoreInterchainAccountResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	hostZone, found := k.GetHostZone(ctx, msg.ChainId)
 	if !found {
-		k.Logger(ctx).Error(fmt.Sprintf("Host Zone not found: %s", msg.ChainId))
+		k.Logger(ctx).Error(fmt.Sprintf("Host Zone not found: %s", msg.ChainId)) //done
 		return nil, types.ErrInvalidHostZone
 	}
 
 	owner := types.FormatICAAccountOwner(msg.ChainId, msg.AccountType)
 
 	// only allow restoring an account if it already exists
-	portID, err := icatypes.NewControllerPortID(owner)
-	if err != nil {
-		errMsg := fmt.Sprintf("could not create portID for ICA controller account address: %s", owner)
-		k.Logger(ctx).Error(errMsg)
-		return nil, err
-	}
+	portID, _ := icatypes.NewControllerPortID(owner)
```

Comment:
this means err have checked in the `icatypes.NewControllerPortID()` function, right?

Reply:
the hostzone check make sure this cannot provide any err. Also in the below function when running RegisterHostzone, this check will also happend again. So no need for this check.